gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/strategies/MapStrategy.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.common.model.impl.properties.strategies;
+
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataInputViewStreamWrapper;
+import org.apache.flink.core.memory.DataOutputViewStreamWrapper;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Strategy class for handling {@code PropertyValue} operations with a value of the type
+ * {@code Map}.
+ */
+public class MapStrategy
+  extends AbstractVariableSizedPropertyValueStrategy<Map<PropertyValue, PropertyValue>> {
+
+  @Override
+  public Map<PropertyValue, PropertyValue> read(DataInputView inputView, byte typeByte)
+      throws IOException {
+    byte[] rawBytes = readVariableSizedData(inputView, typeByte);
+
+    PropertyValue key;
+    PropertyValue value;
+    Map<PropertyValue, PropertyValue> map = new HashMap<>();
+    DataInputViewStreamWrapper internalInputView = createInputView(rawBytes);
+
+    try {
+      while (internalInputView.available() > 0) {
+        key = new PropertyValue();
+        key.read(internalInputView);
+
+        value = new PropertyValue();
+        value.read(internalInputView);
+
+        map.put(key, value);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Error reading PropertyValue", e);
+    }
+
+    return map;
+  }
+
+  @Override
+  public int compare(Map value, Object other) {
+    throw new UnsupportedOperationException(
+      "Method compareTo() is not supported for Map."
+    );
+  }
+
+  @Override
+  public boolean is(Object value) {
+    if (!(value instanceof Map)) {
+      return false;
+    }
+    for (Map.Entry<Object, Object> entry : ((Map<Object, Object>) value).entrySet()) {
```

Comment:
Is that really necessary? ^^

Reply:
I rewrote the cast using wildcards: `Map.Entry<?, ?> entry : ((Map<?, ?>) value).entrySet()`

Reply:
No you misunderstood ... is it possible to have a Map with types other than PropertyValue in this place? Can you model a use-case where the condition in your loop turns true?

Reply:
yes, please see `testCreateWrongParameterizedMap` in the PropertyValueTest-Suite for an example